File: EasyTab/EasyTab.Services/Database/Favourite.cs
using System;
using System.Collections.Generic;

namespace EasyTab.Services.Database;

public partial class Favourite
{
    public int Id { get; set; }

    public int LocaleId { get; set; }

    public int UserId { get; set; }

    public DateTime DateAdded { get; set; }

    public bool IsActive { get; set; }

    public virtual Locale Locale { get; set; } = null!;

    public virtual User User { get; set; } = null!;
}

File: EasyTab/UI/easytab_mobile/lib/models/favourite.dart
import 'package:json_annotation/json_annotation.dart';

part 'favourite.g.dart';

@JsonSerializable()
class Favourite {
  final int? id;
  final int? localeId;
  final int? userId;
  final DateTime? dateAdded;
  final bool? isActive;
  final String? localeName;
  final String? localeLogo;
  final String? localeCategoryName;
  final String? localeCityName;
  final String? localeAddress;

  Favourite({
    this.id,
    this.localeId,
    this.userId,
    this.dateAdded,
    this.isActive,
    this.localeName,
    this.localeLogo,
    this.localeCategoryName,
    this.localeCityName,
    this.localeAddress,
  });

  factory Favourite.fromJson(Map<String, dynamic> json) =>
      _$FavouriteFromJson(json);

  Map<String, dynamic> toJson() => _$FavouriteToJson(this);
}
